panic in llx.arrayContainsAll when containsAll argument evaluates to null (interface conversion: interface {} is nil)

## Summary

`llx.arrayContainsAll` panics with `interface conversion: interface {} is nil, not []interface {}` when the **argument** to `containsAll` evaluates to null. The function nil-guards the bound array (`bind.Value == nil`) but performs an unguarded type assertion on the argument:

```go
// llx/builtin_array.go:934-935
org := bind.Value.([]any)
argFilters := arg.Value.([]any)   // <-- panics when arg.Value == nil
```

Because policy content is server-served, any MQL bundle containing `<array>.containsAll(<expr that can be null>)` can panic cnspec clients fleet-wide across all versions.

## Production impact

This caused incident INC-2026-06-10-client-panic-k8s: the cnspec-enterprise-policies v5.26.0 release rewrote the `mondoo-linux-security` su-restriction check so `suRestrictedGroups` evaluates to null on hosts without `/etc/pam.d/su` (container-optimized k8s node OSes: COS, Flatcar, Bottlerocket). Result: ~10x panic surge (`panic.received`), 1,050+ panics across 19 spaces in ~3h, spanning cnspec 13.22.0, 13.6.0, 13.5.0, and 11.69.1 simultaneously — the bug is latent across major versions and content-triggered.

## Reproduction (verified with released cnspec 13.22.0, build 261db0c0)

In a rootfs **without** `/etc/pam.d/su` (e.g. `ubuntu:24.04` container after `rm /etc/pam.d/su`):

```bash
cnspec run -c '
suRestrictedPam = pam.conf.entries["/etc/pam.d/su"].where(pamType == "auth" && module == "pam_wheel.so" && params["use_uid"] != null)
suRestrictedGroups = suRestrictedPam.where(params["group"] != null).map(params["group"])
suRestrictedPam != empty
suRestrictedGroups == empty || groups.map(name).containsAll(suRestrictedGroups)
suRestrictedGroups != empty || groups.any(name.in(["wheel", "sudo"]))'
```

Panics with:

```
panic: interface conversion: interface {} is nil, not []interface {}

go.mondoo.com/mql/v13/llx.arrayContainsAll(...)
        .../llx/builtin_array.go:935
```

With `/etc/pam.d/su` present, the same query fails cleanly (no panic).

## Expected behavior

A null argument should produce a clean result or query error (e.g. treat null as the bound array's own null handling does: return `RawData{Type, Error}`), never a runtime panic.

## Scope — audit siblings

The same unguarded `arg.Value.([]any)` shape exists in neighboring builtins; at minimum:

- `arrayContainsAll` (`builtin_array.go:935`) — confirmed panicking in production
- `arrayContainsNone` (`builtin_array.go:~988`)
- `arrayContains`/eq variants around `builtin_array.go:~830`, `~873`
- dict variants (`dictContainsAll`, etc.) should be checked for the equivalent hole

--- a/llx/builtin_array.go
+++ b/llx/builtin_array.go
@@ -815,6 +815,14 @@ func arrayDifferenceV2(e *blockExecutor, bind *RawData, chunk *Chunk, ref uint64
 		return nil, rref, err
 	}
 
+	// A null argument (e.g. a missing map key resolving to a typed null
+	// array) propagates as null, mirroring the receiver-nil guard above.
+	// Without this the `arg.Value.([]any)` assertion below panics, which
+	// crashes the whole scan rather than failing the single check.
+	if arg.Value == nil {
+		return &RawData{Type: bind.Type, Error: arg.Error}, 0, nil
+	}
+
 	t := types.Type(arg.Type)
 	if t != bind.Type {
 		return nil, 0, errors.New("called `difference` with wrong type (got: " + t.Label() + ", expected:" + bind.Type.Label() + ")")
@@ -920,6 +928,14 @@ func arrayContainsAll(e *blockExecutor, bind *RawData, chunk *Chunk, ref uint64)
 		return nil, rref, err
 	}
 
+	// A null argument (e.g. a missing map key resolving to a typed null
+	// array) propagates as null, mirroring the receiver-nil guard above.
+	// Without this the `arg.Value.([]any)` assertion below panics, which
+	// crashes the whole scan rather than failing the single check.
+	if arg.Value == nil {
+		return &RawData{Type: bind.Type, Error: arg.Error}, 0, nil
+	}
+
 	t := types.Type(arg.Type)
 	if t != bind.Type {
 		return nil, 0, errors.New("called `arrayNone` with wrong type (got: " + t.Label() + ", expected:" + bind.Type.Label() + ")")
@@ -971,6 +987,14 @@ func arrayContainsNone(e *blockExecutor, bind *RawData, chunk *Chunk, ref uint64
 	arg, rref, err := e.resolveValue(argRef, ref)
 	if err != nil || rref > 0 {
 		return nil, rref, err
+	}
+
+	// A null argument (e.g. a missing map key resolving to a typed null
+	// array) propagates as null, mirroring the receiver-nil guard above.
+	// Without this the `arg.Value.([]any)` assertion below panics, which
+	// crashes the whole scan rather than failing the single check.
+	if arg.Value == nil {
+		return &RawData{Type: bind.Type, Error: arg.Error}, 0, nil
 	}
 
 	t := types.Type(arg.Type)

--- a/providers/core/resources/mql_test.go
+++ b/providers/core/resources/mql_test.go
@@ -741,6 +741,22 @@ func TestArray(t *testing.T) {
 			Code:        "['x'].containsAll([asset.labels['nope']])",
 			Expectation: []any{nil},
 		},
+		// Regression: a null argument (e.g. a missing map key resolving to a
+		// typed null array, as happens on hosts with no /etc/pam.d) must
+		// propagate as null instead of panicking the whole scan with
+		// "interface conversion: interface {} is nil, not []interface {}".
+		{
+			Code:        "a = {a: [1,2]}; [1,2,3].containsAll(a['b'])",
+			Expectation: nil,
+		},
+		{
+			Code:        "a = {a: [1,2]}; [1,2,3].containsNone(a['b'])",
+			Expectation: nil,
+		},
+		{
+			Code:        "a = {a: [1,2]}; [1,2,3] - a['b']",
+			Expectation: nil,
+		},
 		{
 			Code:        "['a','b'] != /c/",
 			ResultIndex: 0, Expectation: true,

--- a/providers/core/resources/mql_test.go
+++ b/providers/core/resources/mql_test.go
@@ -757,6 +757,15 @@ func TestArray(t *testing.T) {
 			Code:        "a = {a: [1,2]}; [1,2,3] - a['b']",
 			Expectation: nil,
 		},
+		// Regression: containsAll of an empty (typed, non-null) list is
+		// vacuously satisfied, so the compiled `== []` check is true. The
+		// mondoo-linux-security su-restriction check relies on this for
+		// `groups.containsAll(suRestrictedGroups)` when no group is configured
+		// and suRestrictedGroups is an empty []string.
+		{
+			Code:        `a = ["x", "y"]; ["wheel", "sudo"].containsAll(a.where(false))`,
+			ResultIndex: 1, Expectation: true,
+		},
 		{
 			Code:        "['a','b'] != /c/",
 			ResultIndex: 0, Expectation: true,

